rasa/nlu/featurizers/sparse_featurizer/count_vectors_featurizer.py

```diff
@@ -798,37 +777,63 @@ def _create_independent_vocab_vectorizers(
     @classmethod
     def load(
         cls,
-        meta: Dict[Text, Any],
-        model_dir: Text,
-        model_metadata: Optional[Metadata] = None,
-        cached_component: Optional["CountVectorsFeaturizer"] = None,
-        should_finetune: bool = False,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
         **kwargs: Any,
-    ) -> "CountVectorsFeaturizer":
+    ) -> CountVectorsFeaturizerGraphComponent:
         """Loads trained component (see parent class for full docstring)."""
-        file_name = meta["file"]
-        featurizer_file = os.path.join(model_dir, file_name)
+        try:
+            with model_storage.read_from(resource) as model_dir:
+                featurizer_file = model_dir / "vocabularies.pkl"
+                vocabulary = io_utils.json_unpickle(featurizer_file)
+
+                share_vocabulary = config["use_shared_vocab"]
 
-        if not os.path.exists(featurizer_file):
-            return cls(meta)
+                if share_vocabulary:
+                    vectorizers = cls._create_shared_vocab_vectorizers(
+                        config, vocabulary=vocabulary
+                    )
+                else:
+                    vectorizers = cls._create_independent_vocab_vectorizers(
+                        config, vocabulary=vocabulary
+                    )
+
+                ftr = cls(
+                    config,
+                    model_storage,
+                    resource,
+                    execution_context,
+                    vectorizers=vectorizers,
+                )
 
-        vocabulary = io_utils.json_unpickle(featurizer_file)
+                # make sure the vocabulary has been loaded correctly
+                for attribute in vectorizers:
+                    ftr.vectorizers[attribute]._validate_vocabulary()
 
-        share_vocabulary = meta["use_shared_vocab"]
+                return ftr
 
-        if share_vocabulary:
-            vectorizers = cls._create_shared_vocab_vectorizers(
-                meta, vocabulary=vocabulary
+        except (ValueError, FileNotFoundError, FileIOException):
```

Comment:
(This reminds me that I definitely shouldn't use ValueErrors for the config validation - otherwise the ValueError here could be from the constructor complaining about a faulty config 😬 .)